compose/config/interpolation.py

```diff
@@ -41,7 +81,9 @@ def interpolate_value(name, config_key, value, section, mapping):
 
 def recursive_interpolate(obj, mapping):
     if isinstance(obj, six.string_types):
-        return interpolate(obj, mapping)
+        if '$' in obj:
+            return interpolate(obj, mapping)
+        return obj
```

Comment:
I guess that doesn't hurt. I think it would be good to move this to the first line of `interpolate()`, that way `interpolate()` is the only function that knows or care about the variable format.  If we were to ever want to change to a different type of variable, the change would be contained to that one function.